patches/tModLoader/Terraria.ModLoader/ModDoubleJump.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+
+namespace Terraria.ModLoader
+{
+	/// <summary>
+	/// A ModDoubleJump instance represents an addition to a Player instance that handles jumping logic. A ModDoubleJump is tied to the player instance, use <see cref="Player.EnableDoubleJump"/> to enable a jump like you would enable any effect through an accessory or armor
+	/// </summary>
+	public class ModDoubleJump
```

Comment:
Yeah not sure if it's a good idea to do this in a separate class, but it seems like a big enough feature for it to be desired. My biggest concern is that it may be confusing to modders as nothing else is really done outside the current ModPlayer class, but since we want to redesign going forward this may not necessarily be an issue.

Reply:
The reasoning behind a separate class is to move all the "control" that `DoubleJumpHooks` does (toggling states of jumping and effects at the appropriate timings which are hardcoded into vanilla) away from the modder so he doesn't have to worry about the specifics. It would take more than 6 `ModPlayer` hooks in proper locations to achieve the same (and they would be redundant for anything other than double jumping), and then the modder would have to handle all the vanilla + mod compatibility himself